gui/src/videodecoder.cpp

```diff
@@ -136,10 +177,14 @@ AVFrame *VideoDecoder::PullFrame()
 		}
 		frame_last = frame;
 		frame = next_frame;
-		int r = avcodec_receive_frame(codec_context, frame);
+		int r = avcodec_receive_frame(codec_context, frame); //r=0 is success
+		
 		if(r == 0)
-		{
-			frame = hw_decode_engine ? GetFromHardware(frame) : frame;
+		{	
+			//ORIG only line: frame = hw_decode_engine ? GetFromHardware(frame) : frame;
+			//frame = hw_decode_engine ? GetFromHardware(frame) : frame;		
```

Comment:
Is it best you do the handling of what to do and not to do when running on an RPi?

My goal has been to lower the CPU use s much as possible which is important to smooth running on it.

I don't know what mechanic you want to use for this?

Reply:
When the pi decoder completely replaces the `VideoDecoder` as mentioned above, this won't be a problem anymore since it will never reach this code.